test/extended/openstack/servers.go

```diff
@@ -172,13 +179,43 @@ func parseInstanceSgs(securityGroups []map[string]interface{}) []string {
 				add = false
 			}
 		}
-		if add == true {
+		if add {
 			result = append(result, fmt.Sprintf("%v", item["name"]))
 		}
 	}
 	return result
 }
 
+// returns the list of addresses
+func getAddressesFromMachine(item objx.Map) []string {
+	listAddresses := objects(item.Get("status.addresses"))
+	result := make([]string, 0, len(listAddresses))
+	for _, item := range listAddresses {
+		if strings.Contains(fmt.Sprintf("%v", item["type"]), "IP") {
+			result = append(result, fmt.Sprintf("%v", item["address"]))
+		}
+	}
+	return result
+}
+
+// parse instance.Addresses object and return a slice of found IPs
+func parseInstanceAddresses(addresses map[string]interface{}) ([]string, error) {
+	result := make([]string, 0, len(addresses)*2)
+	s := fmt.Sprintf("%v", reflect.ValueOf(addresses))
+
+	for _, entry := range strings.Split(s, " ") {
+		if strings.HasPrefix(entry, "addr") {
+			addr := net.ParseIP(strings.Split(entry, ":")[1])
```

Comment:
In case there is no `:`, the slice resulting from `strings.Split(entry, ":")` will have length `1` and indexing `[1]` will panic.

Also, in order to account for IPv6, we should probably use [`strings.SplitN`](https://pkg.go.dev/strings#SplitN):

```Go
	if strings.HasPrefix(entry, "addr") {
		addrLine := strings.SplitN(entry, ":", 2)
		if len(addrLine) > 1 {
			addr := net.ParseIP(addrLine[1])
			if addr == nil {
				return nil, fmt.Errorf("invalid ip format: %v", addrLine[1])
			}
			result = append(result, addr.String())
		}
	}
```

Alternatively, we can leverage [`strings.TrimPrefix`](https://pkg.go.dev/strings#TrimPrefix). `TrimPrefix` returns the string unchanged if the prefix was not found:
```Go
	if addrValue := strings.TrimPrefix(entry, "addr:"); addrValue != entry {
		addr := net.ParseIP(addrValue)
		if addr == nil {
			return nil, fmt.Errorf("invalid ip format: %v", addrValue)
		}
		result = append(result, addr.String())
	}
	return result, nil
```

Reply:
Thanks!! Done